core.impl/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/impl/table/TimeBasedRunPolicy.java

```diff
@@ -0,0 +1,396 @@
+    }
+
+    private TimeRejectionCollection load(final TableKey key)
+    {
+        Statement statement = myGetRejectionsStatement.bind(key.getKeyspace(), key.getTable());
+
+        ResultSet resultSet = mySession.execute(statement);
+        Iterator<Row> iterator = resultSet.iterator();
+        return new TimeRejectionCollection(iterator);
+    }
+
+    @Override
+    public final long validate(final ScheduledJob job)
+    {
+        return -1;
+    }
+
+    @Override
+    public final boolean shouldRun(final TableReference tableReference)
+    {
+        return getRejectionsForTable(tableReference) == -1L;
+    }
+
+    @Override
+    public final void close()
+    {
+        myTimeRejectionCache.invalidateAll();
+        myTimeRejectionCache.cleanUp();
+    }
+
+    /**
+     * Create an instance of Builder class to construct TimeBasedRunPolicy.
+     */
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    /**
+     * Builder class to construct TimeBasedRunPolicy.
+     */
+    public static class Builder
+    {
+        private static final String DEFAULT_KEYSPACE_NAME = "ecchronos";
+
+        private CqlSession mySession;
+        private String myKeyspaceName = DEFAULT_KEYSPACE_NAME;
+        private static long myCacheExpireTime = DEFAULT_CACHE_EXPIRE_TIME;
+        private final Clock myClock = Clock.systemDefaultZone();
+
+        /**
+         * Sets the {@link CqlSession} to be used by the {@link TimeBasedRunPolicy}.
+         *
+         * @param session the {@link CqlSession} to set. Must not be {@code null}.
+         * @return the current {@link Builder} instance for method chaining.
+         */
+        public final Builder withSession(final CqlSession session)
+        {
+            mySession = session;
+            return this;
+        }
+
+        /**
+         * Sets the keyspace name to be used by the {@link TimeBasedRunPolicy}.
+         *
+         * @param keyspaceName the name of the keyspace. Must not be {@code null}.
+         * @return the current {@link Builder} instance for method chaining.
+         */
+        public final Builder withKeyspaceName(final String keyspaceName)
+        {
+            myKeyspaceName = keyspaceName;
+            return this;
+        }
+
+        /**
+         * Builds a new instance of {@link TimeBasedRunPolicy} using the configured parameters.
+         *
+         * @return a new {@link TimeBasedRunPolicy} instance.
+         */
+        public final TimeBasedRunPolicy build()
+        {
+            verifySchemasExists();
+            return new TimeBasedRunPolicy(this);
+        }
+
+        private void verifySchemasExists()
+        {
+            Optional<KeyspaceMetadata> keyspaceMetadata = mySession.getMetadata().getKeyspace(myKeyspaceName);
+            if (!keyspaceMetadata.isPresent())
+            {
+                LOG.error("Keyspace {} does not exist, it needs to be created", myKeyspaceName);
```

Comment:
Creating the same message twice is probably not the best. Do it once, logit and throw it,